curriculum/challenges/english/25-front-end-development/lecture-working-with-relative-and-absolute-units/672bb851b068e3954a05b9b1.md

```diff
@@ -10,6 +10,140 @@ dashedName: what-is-the-calc-function
+
+If there are multiple operands and operators, `calc()` will follow the standard operator precedence rule. You can also add parentheses to establish the order of the operations if needed.
+
+In the example below, you can see a `div` with the text "Hello, World!"
+
+```html
+<div>Hello, World!</div>
+```
+
+Using the CSS type selector for selecting the `div`, you can style it with white text and a dark blue background. 
+
+```css
+div {
+  color: white;
+  background-color: #1b1b32;
+  width: calc(50% - 20px);
+}
+```
+
+What's new here is that the width is calculated dynamically. Notice how we are calling the `calc()` function and passing an expression as an argument. The expression has two operands with different units and one operator, the subtraction operator.
+
+Percentage is a relative unit. The value (50%) will be determined by the width of the parent container. Then, 20px is subtracted from the value. The result of this expression will determine the width of the `div`.
+
+The width of the `div` is approximately half the total width of its container, and if you resize the parent container, the width will be recalculated automatically.
+
+That's the key advantage of using `calc()`. You can determine the value of a CSS property dynamically based on different aspects of the application or viewport.
+
+The expression can also contain CSS functions and variables if you need to use them in your calculations. You'll learn more about CSS variables in the next lectures.
+
+Great. Now that you know about the basics of the `calc()` function, let's cover some of its best practices.
+
+First, you must surround the addition (+) and subtraction (-) operators with whitespace.
+
+For example, this would not be a valid expression because the subtraction operator is immediately before the second operand.
+
+```css
+calc(100% -30px)
+```
+
+The subtraction (-) operator must be surrounded by whitespace, like this. Adding the whitespace will create a valid expression.
+
+```css
+calc(100% - 30px)
+```
+
+This is not necessary for the multiplication and division operators but it's highly recommended.
+
+You can also nest `calc()` function calls if you need to perform calculations and use those results in other calculations.
+
+Also, if you use the value zero to represent length in the expression that you pass into the `calc()` function, you must include the units.
+
+For example, this expression would not be valid:
+
+```css
+calc(100% - 0)
+```
+
+You would need to add the units, like px.
+
+```css
+calc(100% - 0px)
+```
+
+You should also know that currently, if you use the multiplication or division operators, one of the operands has to be unitless. 
+
+For the division operator, specifically the right operand has to be unitless.
+
+This would not be a valid expression because both operands have units (pixels). One of the operands, either 5 or 50, must be unitless.
```

Comment:
```suggestion
You should also know that currently, if you use the multiplication or division operators, one of the operands has to be unitless. For the division operator, specifically the right operand has to be unitless. This would not be a valid expression because both operands have units (pixels). One of the operands, either 5 or 50, must be unitless:
```